scripts/irods/test/test_quotas.py

```diff
@@ -131,25 +131,94 @@ def acRescQuotaPolicy(rule_args, callback, rei):
                     self.user0.assert_icommand(['icp', '-R', resc_name, file_name, f'{file_name}_case_11'])
                     self.admin.assert_icommand(['iadmin', 'cu'])
+                    self.admin.assert_icommand(['iadmin', 'lq'], 'STDOUT', ['quota_over: 3192', 'quota_over: -1808'])
 
                 # Case 12: should fail with just a single quota violated
                 with self.subTest(testcase=12, desc="Violating quota"):
                     self.user0.assert_icommand(['icp', '-R', resc_name, file_name, f'{file_name}_case_12'], 'STDERR_SINGLELINE', 'SYS_RESC_QUOTA_EXCEEDED')
+                    self.admin.assert_icommand(['iadmin', 'lq'], 'STDOUT', ['quota_over: 3192', 'quota_over: -1808'])
         finally:
             self.admin.run_icommand(['irm', '-f', f'{file_name}_case_10', f'{file_name}_case_11', f'{file_name}_case_12'])
             self.admin.run_icommand(['iadmin', 'rmresc', resc_name])
             self.admin.run_icommand(['iadmin', 'rmgroup', f'{group_name}_1'])
             self.admin.run_icommand(['iadmin', 'rmgroup', f'{group_name}_2'])
             IrodsController().reload_configuration()
+
+    # The following test covers case 40, 41, 42
+    def test_physical_resource_and_total_quota_simultaneously__issue_4089(self):
+        pep_map = {
+            'irods_rule_engine_plugin-irods_rule_language': textwrap.dedent('''
+                acRescQuotaPolicy {
+                    msiSetRescQuotaPolicy("on");
+                }
+            '''),
+            'irods_rule_engine_plugin-python': textwrap.dedent('''
+                def acRescQuotaPolicy(rule_args, callback, rei):
+                    callback.msiSetRescQuotaPolicy('on')
+            ''')
+        }
+
+        try:
+            with temporary_core_file() as core:
+                core.add_rule(pep_map[self.plugin_name])
+                IrodsController().reload_configuration()
+
+                group_name = 'test_group_case_40_41_42'
+                file_name = 'test_file_case_40_41_42'
+                resc_name = 'ufs0_quota_case_40_41_42'
+                lib.make_file(os.path.join(self.user0.local_session_dir, file_name), 4096, contents='arbitrary')
+                lib.create_ufs_resource(self.admin, resc_name)
+
+                self.admin.assert_icommand(['iadmin', 'mkgroup', f'{group_name}_1'])
+                self.admin.assert_icommand(['iadmin', 'sgq', f'{group_name}_1', resc_name, '5000'])
+                self.admin.assert_icommand(['iadmin', 'sgq', f'{group_name}_1', 'total', '15000'])
+
+                self.admin.assert_icommand(['iadmin', 'atg', f'{group_name}_1', self.user0.username])
+
+                self.user0.assert_icommand(['iput', '-R', self.testresc, os.path.join(self.user0.local_session_dir, file_name)])
+                self.admin.assert_icommand(['iadmin', 'cu'])
+                self.admin.assert_icommand(['iadmin', 'lq'], 'STDOUT', ['quota_over: -5000', 'quota_over: -10904'])
```

Comment:
For all `iadmin lq` assertions, consider including newlines, etc. to delimit substrings. It would help protect against false positives, etc.

For example, the following guarantees that -5000 is matched rather than -50001. Same thing for the beginning of the label (quota_over), etc.
```py
self.admin.assert_icommand(['iadmin', 'lq'], 'STDOUT', ['\nquota_over: -5000\n', '\nquota_over: -10904\n'])
```